### DISCUSSION TITLE
using ternary in default value controlled usefieldArray

### DISCUSSION THREAD

**Discussion by tempoeric** [react-hook-form/react-hook-form#6805] [Q&A]
Hi im using useFieldArray controlled but i cannot add ternary in the default value section why is that
![image](https://user-images.githubusercontent.com/42777871/137274356-0b51ac3f-530f-40e7-81ee-078350610b92.png)

**Comment by tempoeric** [comment]
@bluebill1049 please help

**Reply by tempoeric** [reply]
@bluebill1049 please provide an answer

**Comment by thanh-nguyen-95** [comment]
Please prodvide a CodeSandBox so we can help you better. We need to know which version of React Hook Form you're using, how you structure your component, etc,

**Comment by tempoeric** [comment]
HI @thanh-nguyen-95  and @bluebill1049 here is a sample codebox of what im trying to achieve i want to use ternary into defaultvalue so it will know which is trying to render if its ph it will use the field?.city else field.cityText 
https://codesandbox.io/s/flamboyant-hofstadter-6luyx?file=/src/BeneficialOwner.tsx

**Reply by thanh-nguyen-95** [reply]
Please make sure the CodeSandbox is at least working, which shows the UI with any type of error or unexpected value. It doesn't even render the app itself. Also it would be great if you can add some default values to the form.

I'm sorry to say this but you cannot expect us to help you with such a half-baked CodeSandBox.

**Reply by tempoeric** [reply]
sorry but my issue only is the default value not on rendering the UI if you read the code you can easily understand what im trying to do

**Comment by thanh-nguyen-95** [comment]
> https://codesandbox.io/s/flamboyant-hofstadter-6luyx?file=/src/BeneficialOwner.tsx

If your code is the same as the CodeSandbox, I spot a few things that might make it work:

- The app is not rendered
- `watch` is not passed in to `BeneficialOwnerList` as a prop
- Each field should have a `key` prop preferably assigned to `field.id`
- [`getValues`](https://react-hook-form.com/v6/api#getValues) can be used instead of `watch`. I think it's better for performance

That's all I can tell right now. Try and see if it works. I wrote this in a seperate comment so it can be marked as an answer

**Reply by tempoeric** [reply]
@thanh-nguyen-95  @bluebill1049  here is an updated and working sandbox https://codesandbox.io/s/fervent-matsumoto-bt4u0?file=/src/Beneficiary/index.tsx please check why i dont have default value in the first render of the dropdowns but when i change the value of country it renders the default value
